type.go

```diff
@@ -278,6 +285,18 @@ func (t booleanType) Decode(dst encoding.Values, src []byte, enc encoding.Encodi
 	return encoding.DecodeBoolean(dst, src, enc)
 }
 
+func (t booleanType) AssignValue(dst reflect.Value, src Value) error {
+	v := src.Boolean()
+	switch dst.Kind() {
+	case reflect.Bool:
+		dst.SetBool(v)
+		return nil
+	}
+
+	dst.Set(reflect.ValueOf(v))
+	return nil
+}
```

Comment:
Maybe rewrite these methods to have only one `return nil` instead of having to return in each case?
```suggestion
	v := src.Boolean()
	switch dst.Kind() {
	case reflect.Bool:
		dst.SetBool(v)
	default:
		dst.Set(reflect.ValueOf(v))
	}
	return nil
}
```

Reply:
Good idea. Updated all relevant switches in 6119f39